docs/react-v9/contributing/rfcs/react-components/convergence/extended-design-tokens.md

```diff
@@ -0,0 +1,455 @@
+| Prefix    | Prefix to distinguish control-specific tokens from semantic tokens that might share the same name, e.g. Button. Only applies to control tokens | ctrl / smtc                                                    |
+| Component | The base UI component or component family, aka semantic group                                                                                  | Button<br>Switch                                               |
+| Variant   | A variant of the base component for more specific use cases                                                                                    | Secondary<br>Compound<br>Brand                                 |
+| Part      | A specific element within the default component. Can be a single part or multiple parts. Typically maps to a DOM element                       | Icon<br>SecondaryIcon                                          |
+| Property  | A visual attribute of the Component or Part. Often maps to a CSS property, but not always (e.g. stroke)                                        | ForegroundColor<br>BackgroundColor<br>IconAfter<br>StrokeWidth |
+| Modifier  | Context that changes the value of a property based on state or mode. Usually maps to an interactive state or mode.                             | Rest<br>Hover<br>Pressed<br>Disabled<br>Selected-\*            |
+
+_Variants_
+
+The term “variants” is often used to describe alternative forms or styles of components. There are typically 3 types of variants representing different axes of changes:
+
+1.  **Style variant**: Component has a different look and feel, but same behaviors and visual structure as the base component.
+
+    a. In code, v9 typically uses appearance prop
+
+    b. In Figma, the Style property controls these
+
+    c. Example: Button has Secondary, Primary, Subtle, Transparent style variants
+
+2.  **Layout variant**: Component has a different visual structure from the base component, but typically retains the same behaviors and data. Different Parts may be toggled on/off or have different visual treatments due to their composition.
+
+    a. In code, this might be controlled through optional component props e.g. Button icon and iconPosition
+
+    b. In Figma,
+
+    c. Example: Buttons can have just a text label, just an icon, or an icon and Button
+
+3.  **Functional variant**: Component exposes different interactions or data than the base component.
+
+    a. In code, these might be separate components with different APIs
+
+## Expose expanded tokens in Theme
+
+Each component that exposes tokens would add a Component.tokens.ts file to expose its tokens API:
+
+```javascript
+// Button.tokens.ts
+
+import { tokens } from '@fluentui/react-theme';
+
+// Semantic button, shared between all Button-family components
+
+export const buttonGroupTokens = {
+
+buttonBorderRadius: \`var(--buttonBorderRadius, ${tokens.borderRadiusMedium})\`,
```

Comment:
If this is the final design, how we are going to control typo and implementation: `var(--buttonBrderRadius, ${tokens.borderRadiusMedium})` - "o" is missing and the whole thing is broken and impossible to debug due an amount of tokens. Will it be automatically generated?

Reply:
It will be manually added, run through an LLM as a unit test and reviewed. We'll release it one component at a time